sudospawner/mediator.py

```diff
@@ -50,12 +49,19 @@ def kill(pid, signal):
     app_log.debug("Sending signal %i to %i", signal, pid)
     try:
         os.kill(pid, signal)
-    except OSError as e:
-        if e.errno == errno.ESRCH:
-            # not running
-            alive = False
-        else:
-            raise
+    except ProcessLookupError:
+        # not running
+        alive = False
+    except PermissionError:
```

Comment:
Also, I'm OK if you want unexpected exceptions to raise so that they get reported as bugs instead of lost in logs. I'm on the fence.